src/binfhe/include/binfhecontext.h

```diff
@@ -156,6 +182,38 @@ class BinFHEContext : public Serializable {
     LWECiphertext Encrypt(ConstLWEPrivateKey sk, const LWEPlaintext& m, BINFHE_OUTPUT output = BOOTSTRAPPED,
                           LWEPlaintextModulus p = 4, NativeInteger mod = 0) const;
 
+    /**
+   * Encrypts a bit using a public key (public key encryption)
+   *
+   * @param pk - the public key
+   * @param &m - the plaintext
+   * @param p - plaintext modulus
+   * @param mod Encrypt according to mod instead of m_q if mod != 0
+   * @param output - SMALLN to generate ciphertext with dimension n (default)
+   * @return a shared pointer to the ciphertext
+   */
+    LWECiphertext Encrypt(ConstLWEPublicKey pk, const LWEPlaintext& m, BINFHE_OUTPUT output = SMALLN,
+                          LWEPlaintextModulus p = 4, NativeInteger mod = 0) const;
+    /**
+   * Encrypts a bit using a public key (public key encryption) with larger dimension N and modulus Q
+   *
+   * @param pk - the public key
+   * @param &m - the plaintext
+   * @param p - plaintext modulus
+   * @param mod Encrypt according to mod instead of m_q if mod != 0
+   * @return a shared pointer to the ciphertext
+   */
+    LWECiphertext EncryptN(ConstLWEPublicKey pk, const LWEPlaintext& m, LWEPlaintextModulus p = 4,
```

Comment:
passing m by const reference may seems as a good decision, but it is needless. LWEPlaintext is int64_t and should be passed by value. in this case you don't need to make it const and passing by value may be even faster.
on the contrary, "NativeInteger mod" should be passed by const reference as it is an object of a custom class.